main.cpp

```diff
@@ -180,139 +123,268 @@ int main()
 
 #include <iostream>
 
+struct DoubleType;
+struct IntType;
+
 struct FloatType
 {
-    float add(float lhs, float rhs);
-    float subtract(float lhs,float rhs );
-    float multiply(float lhs, float rhs );
-    float divide(float lhs,float rhs );
+    float value;
+
+    FloatType& add( float rhs );
```

Comment:
This does not demonstrate the following:
```
Edit your 3 structs so that they own a heap-allocated primitive type 
without using smart pointers named 'value'
```
- [ ] revise